rfcs/0003-image-filter.md

```diff
@@ -0,0 +1,176 @@
+mg-test-image5   True      COMMIT            gcr.io/cf-build-service-dev-219913/test/mg-test-image5@sha1:EF305A16C96FF55AA3A9A764FD942D6DDA5B34AF    app-team
+...
+```
+
+------------------------------
+
+**Print Image Configs With A Certain Reason**
+
+```
+$kp image list --latest-reason stack -A
+
+NAME             READY     LATEST REASON     LATEST IMAGE                                                                                            NAMESPACE   
+ 
+mg-test-image1   False     STACK                                                                                                                     mg-test
+mg-test-image2   True      STACK             gcr.io/cf-build-service-dev-219913/test/mg-test-image2@sha1:B75D2E59189B813285726A36CD0A737D2720832C    some-ns
+mg-test-image3   False     STACK                                                                                                                     workspace
+mg-test-image4   True      STACK             gcr.io/cf-build-service-dev-219913/test/mg-test-image4@sha1:B2F2F07F4583C3870131B8A3447C7D04B046A0E5    acceptance
+mg-test-image5   False     STACK                                                                                                                     app-team
+... 
+```
+
+------------------------------
+
+**Print Image Configs That Are Ready And Have A Stack or Buildpack Latest Reason**
+
+```
+$kp image list --status ready --latest-reason stack, buildpack -A
+
+NAME              READY     LATEST REASON     LATEST IMAGE                                                                                             NAMESPACE                                                                                             
+
+mg-test-image1    True      STACK             gcr.io/cf-build-service-dev-219913/test/mg-test-image@sha1:13D5582884803A7755CEEDCB7478485498C4F1C4      mg-test  
+mg-test-image2    True      BUILDPACK         gcr.io/cf-build-service-dev-219913/test/mg-test-image2@sha1:2E46CC4B032552F76ADD7622F63D953F330C428A     some-ns  
+mg-test-image3    True      STACK             gcr.io/cf-build-service-dev-219913/test/mg-test-image3@sha1:32EAA227FBEC2CBF6C1D3C3A84062FB2C41A55D1     workspace 
+mg-test-image4    True      BUILDPACK         gcr.io/cf-build-service-dev-219913/test/mg-test-image4@sha1:B2F2F07F4583C3870131B8A3447C7D04B046A0E5     acceptance 
+mg-test-image5    True      STACK             gcr.io/cf-build-service-dev-219913/test/mg-test-image5@sha1:EF305A16C96FF55AA3A9A764FD942D6DDA5B34AF     app-team 
+...
+```
+
+------------------------------
+
+**Image Configs Are Not Ready Because Adding A New Buildpack To The Default Store Caused a Breaking Change**
+
+```   
+$kp image list --clusterbuilder default -A
+
+NAME             READY      LATEST REASON     LATEST IMAGE     NAMESPACE                                                                                         
+
+mg-test-image1   False      BUILDPACK                          matt-fun-zone
```

Comment:
We currently sort images by creation time whereas `kubectl` sorts by namespace first then alphabetically. Do we want to match that? Or maybe do namespace first then creation time

Without namespace sorting it might be harder to parse.

Reply:
@tylerphelan @sukhil-suresh @tomkennedy513 I agree with the suggestion, but I'm wondering if it makes more sense for that detail to be addressed in one of the stories that implements the RFC.  I'm concerned about being too prescriptive at the RFC level, imo stories should be broken off of it not serve as the story for implementation.